tools/nightly_checkout.py

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env python3
+"""Checks out the nightly development version of PyTorch and installs pre-built
+binaries into the repo.
+
+You can use this script to check out a new nightly branch with the following::
+
+    $ ./tools/nightly_checkout.py -b my-nightly-branch
+    $ conda activate pytorch-deps
+
+Or if you would like to re-use an existing conda environment, you can pass in
+the regular environment parameters (--name or --prefix)::
+
+    $ ./tools/nightly_checkout.py -b my-nightly-branch -n my-env
+    $ conda activate my-env
+"""
+import os
+import re
+import sys
+import json
+import glob
+import time
+import shutil
+import logging
+import tempfile
+import functools
+import contextlib
+import subprocess
+from ast import literal_eval
+from argparse import ArgumentParser
+
+
+LOGGER = logging.getLogger("conda-pytorch")
+URL_FORMAT = "{base_url}/{platform}/{dist_name}.tar.bz2"
+SHA1_RE = re.compile("([0-9a-fA-F]{40})")
+SPECS_TO_INSTALL = ("pytorch", "mypy", "pytest", "hypothesis", "ipython", "sphinx")
+
+
+def init_logging(level=logging.INFO):
+    """Start up the logger"""
+    logging.basicConfig(level=level)
```

Comment:
You're probably going to end up spending some time debugging end user issues, especially from users who are not technically savvy. Might I suggest you consider adding a "rage" based mechanism? At Facebook, "rage" refers to the capability to retroactively get the logs from a run that failed, rather than having to ask the user to rerun the command again with logging. I have a simple implementation at https://github.com/ezyang/ghstack/blob/master/ghstack/logging.py

Reply:
Thanks for the link.  I'll boost the logging game here

Reply:
I have added logging based on what you had here. In the process I noticed some issues with conda v4.8.4 not being able to perform the solve.  Let me know if you encounter that.  I downgraded to v4.8.3